.github/scripts/trymerge.py

```diff
@@ -652,6 +654,25 @@ def merge_into(self, repo: GitRepo, *, force: bool = False, dry_run: bool = Fals
 
         repo.push(self.default_branch(), dry_run)
 
+    def merge_on_green(self, repo: GitRepo, org: str, project: str, pr_num: int, *, dry_run: bool = False) -> None:
+        start_time = time.time()
+
+        while True:
+            current_time = time.time()
+            elapsed_time = current_time - start_time
+
+            if(elapsed_time > 355 * 60):
+                msg = 'Took too long to merge. Please try again later'
+                gh_post_comment(org, project, pr_num, msg, dry_run=dry_run)
+                gh_add_labels(org, project, pr_num, ["land-failed"])
+                raise RuntimeError(msg)
+
+            try:
+                self.merge_into(repo, dry_run=dry_run)
+            except Exception as e:
+                print(f"Merge failed due to {e}. Trying again in 60 seconds.")
```

Comment:
Shouldn't we then check if it failed for a legitimate reason? I think this is a good place to parse the reason and if it's something other than waiting for status checks, it should fail and not keep while looping.

Reply:
oh good point. I guess we only need to try again if it's a check exception and stop if there's any other types.